File: 프로그래머스/Lv.2/42578. 의상/의상.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dic = [String:Int]()
    var answer = 1
    for c in clothes {
        dic[c[1], default: 0] += 1
    }
    for d in dic {
        answer *= d.value + 1
    }
    return answer - 1
}

File: 프로그래머스/lv2/의상.py
a = [["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]]

def solution(clothes):
    answer = 1
    dic = {}
    for cloth in clothes:
        dic[cloth[1]] = dic.get(cloth[1], 0) + 1

    for key in dic:
        answer *= dic[key] + 1
        
    return answer - 1

print(solution(a))


